syntax = "proto3";

package feed;

// Feed Item Message
message FeedItem {
    string id = 1;
    int32 post_id = 2;
    int32 user_id = 3;
    string created_at = 4;
    string updated_at = 5;
}

// Feed Item List Message
message FeedItemList {
    repeated FeedItem feed_items = 1;
}

// Get Feed Request Message
message GetFeedRequest {
    int32 user_id = 1;
    int32 limit = 2;
    int32 offset = 3;
}

// Feed Response Message
message FeedResponse {
    bool success = 1;
    string message = 2;
    FeedItem feed_item = 3;
}

// Get Feed Response Message
message GetFeedResponse {
    bool success = 1;
    string message = 2;
    FeedItemList feed_items = 3;
}

// Feed Service Definition
service FeedService {
    rpc GetFeed(GetFeedRequest) returns (GetFeedResponse) {}
    rpc AddToFeed(FeedItem) returns (FeedResponse) {}
    rpc RemoveFromFeed(FeedItem) returns (FeedResponse) {}
} 